controllers/concept.go

```diff
@@ -187,23 +188,45 @@ func (u ConceptController) GetFilteredConceptRows(sourceId int, cohortId int, co
 	for idx, conceptId := range conceptIds {
 		// run each query with a longer list of filterConceptIds, until the last query is run with them all:
 		filterConceptIds := conceptIds[0 : idx+1]
-		breakdownStats, err := u.conceptModel.RetrieveBreakdownStatsBySourceIdAndCohortIdAndConceptIds(sourceId, cohortId, filterConceptIds, breakdownConceptId)
+		// use empty cohort pairs list:
+		filterCohortPairs := [][]int{}
+		breakdownStats, err := u.conceptModel.RetrieveBreakdownStatsBySourceIdAndCohortIdAndConceptIdsAndCohortPairs(sourceId, cohortId, filterConceptIds, filterCohortPairs, breakdownConceptId)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve concept Breakdown for concepts %v due to error: %s", filterConceptIds, err.Error())
+		}
+
+		conceptValuesToPeopleCount := getConceptValueToPeopleCount(breakdownStats)
+		variableName := conceptIdToName[conceptId]
+		log.Printf("Generating row for variable with name %s", variableName)
+		generatedRow := generateRowForVariable(variableName, conceptValuesToPeopleCount, sortedConceptValues)
+		rows = append(rows, generatedRow)
+	}
+
+	return rows, nil
+}
+
+func (u ConceptController) GetCustomDichotomousVariablesAttritionRows(sourceId int, cohortId int, filterConceptIds []int64, filterCohortPairs [][]int, breakdownConceptId int64, sortedConceptValues []string) ([][]string, error) {
+	var rows [][]string
+	for idx, cohortPair := range filterCohortPairs {
+		// run each query with the full list of filterConceptIds and an increasingly longer list of filterCohortPairs, until the last query is run with them all:
```

Comment:
i see that the chunk of code between line 211-221 is identical to line 190-202 in `GetConceptVariablesAttritionRows`, not sure if we should make a helper function since the number of parameters might be too large but just wanted to bring this up.

Reply:
they are similar and could probably be merged. Maybe when we get to the part where there is a requirement to generate the rows in the same specific order as seen in the UI. For now I think it is OK to leave them like this. I'll add a TODO in the code with this comment.